File: native/ios/HybridApp/Bridge/Plugins/ScanPlugin.swift
import AVFoundation

/// 扫码插件：二维码 / 条形码
final class ScanPlugin: BridgePlugin {
    var bridge: JSBridgeContentController?
    let namespace = "device.scan"

    func handle(method: String, params: Any, completion: @escaping (BridgeResult) -> Void) {
        switch method {
        case "scanCode":
            // 实际项目接入自研扫码页或第三方扫码 SDK
            completion(BridgeResult(code: 0, msg: "ok", data: ["result": "https://example.com/qr"]))
        default:
            completion(BridgeResult(code: -1, msg: "method not found", data: nil))
        }
    }
}

File: native/android/app/src/main/java/com/example/hybrid/bridge/plugins/ScanPlugin.kt
package com.example.hybrid.bridge.plugins

import android.content.Context
import com.example.hybrid.bridge.BridgePlugin
import com.example.hybrid.bridge.BridgeResult
import org.json.JSONObject

class ScanPlugin(private val context: Context) : BridgePlugin {
    override val namespace: String = "device.scan"

    override fun handle(method: String, params: JSONObject, completion: (BridgeResult) -> Unit) {
        when (method) {
            "scanCode" -> {
                // 实际项目接入 ZXing 或自研扫码页
                completion(BridgeResult(0, "ok", JSONObject().put("result", "https://example.com/qr")))
            }
            else -> completion(BridgeResult(-1, "method not found", null))
        }
    }
}
